src/value_ptr.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#ifndef CATA_VALUE_PTR_H
+#define CATA_VALUE_PTR_H
+
+#include <memory>
+
+namespace cata
+{
+/**
+ * This class is essentially a copyable unique pointer. Its purpose is to allow
+ * for sparse storage of data without wasting memory in classes such as itype.
+ */
+template <class T>
+class value_ptr : public std::unique_ptr<T>
+{
+    public:
+        value_ptr() = default;
+        value_ptr( T *value ) : std::unique_ptr<T>( value ) {}
+        value_ptr( const value_ptr<T> &other ) :
```

Comment:
I think you need an explicitly-defaulted move constructor too.  If you want to double-check, you could add a unit test that a moved-from `value_ptr` is empty (which would be a good thing to have in any case).

Reply:
you were totally right @jbytheway, I defaulted the move constructor and added tests for all of the functions.